PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -511,18 +719,19 @@ void ProjectPanel::openSelectFile()
 	TVITEM tvItem;
 	tvItem.mask = TVIF_PARAM;
 	tvItem.hItem = _treeView.getSelection();
-	::SendMessage(_treeView.getHSelf(), TVM_GETITEM, 0,(LPARAM)&tvItem);
+	::SendMessage(_treeView.getHSelf(), TVM_GETITEM, 0,reinterpret_cast<LPARAM>(&tvItem));
 
 	NodeType nType = getNodeType(tvItem.hItem);
-	generic_string *fn = (generic_string *)tvItem.lParam;
-	if (nType == nodeType_file && fn)
+	ProjectPanelData& data = *reinterpret_cast<ProjectPanelData*>(tvItem.lParam);

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `FALSE` before using `tvItem.lParam`.